src/install/mod.rs

```diff
@@ -0,0 +1,245 @@
+//! Functionality related to installing prebuilt binaries and/or running cargo install.
+
+use self::krate::Krate;
+use binary_install::{Cache, Download};
+use child;
+use emoji;
+use failure::{self, ResultExt};
+use log::debug;
+use log::{info, warn};
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+use std::process::Command;
+use target;
+use which::which;
+use PBAR;
+
+mod krate;
+mod mode;
+mod tool;
+pub use self::mode::InstallMode;
+pub use self::tool::Tool;
+
+/// Install a cargo CLI tool
+///
+/// Prefers an existing local install, if any exists. Then checks if there is a
+/// global install on `$PATH` that fits the bill. Then attempts to download a
+/// tarball from the GitHub releases page, if this target has prebuilt
+/// binaries. Finally, falls back to `cargo install`.
+pub fn install(
```

Comment:
Perhaps `install` => `cargo_install_or_download_prebuilt`?

(to emphasize that this only works for cargo-based tools due to the fallback)

Reply:
ah yeah very good call. when we merge #625 with this that'll be important.